### ISSUE TITLE
After upgrading to 0.5.0 - SSR component doesn't see the props

### ISSUE THREAD

**Issue by maximblack** [apollostack/react-apollo#199]
On 0.4.7 works fine, but after upgrading to 0.5.0 or 0.5.1 on server side rendering i am getting:

<details>
```js
TypeError: Cannot read property 'data' of undefined
   at Profile.render (/media/maximblack/hdd/projects/dada/ui/components/Profile.js:46:12)
   at getChildFromComponent (/media/maximblack/hdd/projects/dada/node_modules/react-apollo/server.js:16:26)
   at getQueriesFromTree (/media/maximblack/hdd/projects/dada/node_modules/react-apollo/server.js:43:24)
   at getQueriesFromTree (/media/maximblack/hdd/projects/dada/node_modules/react-apollo/server.js:42:9)
   at /media/maximblack/hdd/projects/dada/node_modules/react-apollo/server.js:49:76
   at forEachSingleChild (/media/maximblack/hdd/projects/dada/node_modules/react/lib/ReactChildren.js:52:8)
   at traverseAllChildrenImpl (/media/maximblack/hdd/projects/dada/node_modules/react/lib/traverseAllChildren.js:69:5)
   at traverseAllChildrenImpl (/media/maximblack/hdd/projects/dada/node_modules/react/lib/traverseAllChildren.js:85:23)
   at traverseAllChildren (/media/maximblack/hdd/projects/dada/node_modules/react/lib/traverseAllChildren.js:164:10)
   at Object.forEachChildren [as forEach] (/media/maximblack/hdd/projects/dada/node_modules/react/lib/ReactChildren.js:72:3)
   at getQueriesFromTree (/media/maximblack/hdd/projects/dada/node_modules/react-apollo/server.js:49:26)
   at getQueriesFromTree (/media/maximblack/hdd/projects/dada/node_modules/react-apollo/server.js:42:9)
   at /media/maximblack/hdd/projects/dada/node_modules/react-apollo/server.js:49:76
   at forEachSingleChild (/media/maximblack/hdd/projects/dada/node_modules/react/lib/ReactChildren.js:52:8)
   at traverseAllChildrenImpl (/media/maximblack/hdd/projects/dada/node_modules/react/lib/traverseAllChildren.js:69:5)
   at traverseAllChildrenImpl (/media/maximblack/hdd/projects/dada/node_modules/react/lib/traverseAllChildren.js:85:23)
   at traverseAllChildren (/media/maximblack/hdd/projects/dada/node_modules/react/lib/traverseAllChildren.js:164:10)
   at Object.forEachChildren [as forEach] (/media/maximblack/hdd/projects/dada/node_modules/react/lib/ReactChildren.js:72:3)
   at getQueriesFromTree (/media/maximblack/hdd/projects/dada/node_modules/react-apollo/server.js:49:26)
   at getQueriesFromTree (/media/maximblack/hdd/projects/dada/node_modules/react-apollo/server.js:42:9)
   at getQueriesFromTree (/media/maximblack/hdd/projects/dada/node_modules/react-apollo/server.js:42:9)
   at getQueriesFromTree (/media/maximblack/hdd/projects/dada/node_modules/react-apollo/server.js:42:9)
   at getQueriesFromTree (/media/maximblack/hdd/projects/dada/node_modules/react-apollo/server.js:42:9)
   at getDataFromTree (/media/maximblack/hdd/projects/dada/node_modules/react-apollo/server.js:61:14)
   at /media/maximblack/hdd/projects/dada/api/index.js:195:2
   at /media/maximblack/hdd/projects/dada/api/index.js:173:26
   at /media/maximblack/hdd/projects/dada/node_modules/react-router/lib/match.js:65:5
   at /media/maximblack/hdd/projects/dada/node_modules/react-router/lib/createTransitionManager.js:118:11
```
</details>

If i run it in webpack-dev-server, it's ok

**Comment by jbaxleyiii** [comment]
@maximblack the 0.5.0 removed the manual call to componentDidMount on the server. I'm working to write a custom server renderer that supports it correctly (mainly the setState). However, looking at https://github.com/facebook/react/issues/7671 it may be a bad pattern anyway

**Comment by jbaxleyiii** [comment]
@maximblack could you provide what your component looks like?

**Comment by maximblack** [comment]
It's simple `this.props` call. And the `this.props` is `undefined`. Tried in constructor and render method of component. So the component doesn't receive props.

**Comment by jbaxleyiii** [comment]
@maximblack that certainly seems like a bug. Could you try to create a failing test case? We have SSR tests that are all passing so a failing test would be critical for me to know what is wrong

**Comment by helfer** [comment]
@jbaxleyiii I think I see the same bug in https://github.com/apollostack/GitHunt-React/issues/89

**Comment by jbaxleyiii** [comment]
Yep, I'll work on a fix today!

**Comment by jbaxleyiii** [comment]
@helfer do you think you could create a failing test case?

**Comment by helfer** [comment]
I'll take a look. Maybe @davidyaha can help as well.

**Comment by maximblack** [comment]
@helfer yes, it's the same issue i think. I also encountered when using GitHunt project. Tried with apollo-client@0.4.13 and with the last version of react-apollo and works fine

**Comment by helfer** [comment]
@maximblack are you saying it works with apollo-client 0.4.13 and the newest react-apollo, but not with apollo-client 0.4.14 and the newest react-apollo?

**Comment by jbaxleyiii** [comment]
@maximblack @helfer this is the commit that introduced the regression. Namely line 49 in `src/server.ts`. I'd like to keep the `componentWillMount()` call out until I can write a custom renderer that supports transactions correctly.

The server tests are super minimal and the first step in my opinion is a few more tests (failing or now) around react server lifecycle events. Namely `constructor, componentWillMount, and render`.

I'll happily fix this in the next couple of days (trying to do better about resting on the weekends 🎉 ) but a few tests would go a long way if anyone can help with it!

Thanks!

**Comment by jbaxleyiii** [comment]
fixed with #205 which will be out shortly
